### PR TITLE
Fixes #32309 - pulpcore-manager fails from certain directories

### PR DESCRIPTION
**Pull request by ianballou**
Sets a specific directory for pulpcore-manager to be called from.

### CODE REVIEW

File: manifests/admin.pp [hunk 1]
```diff
@@ -32,11 +35,13 @@
   String $user = $pulpcore::user,
   Stdlib::Absolutepath $pulp_settings = $pulpcore::settings_file,
   Optional[Integer[0]] $timeout = undef,
+  String $working_dir = $pulpcore::user_home,
```
Comment (ekohl) [inline manifests/admin.pp:38]: I prefer to be stricter, even if it doesn't really make a difference.
```suggestion
  Stdlib::Absolutepath $working_dir = $pulpcore::user_home,
```

**Review by ekohl — COMMENTED** [review]
The change itself makes sense to me. However, please make the commit message more descriptive. Like `Always run admin cmds from Pulp's home` (not sure how much will fit). Ideally with then a body that describes that the cwd might be an unreadable directory like `/root` and dynaconf appears to unconditionally read the current directory. Pulp's user home should always be readable for the user pulp.

Also, it might be good to link to a dynaconf bugreport. IMHO it shouldn't error out on unreadable CWD.

**Comment by ianballou** [comment]
Not sure yet why the tests are complaining about `Unknown variable: 'pulpcore::user_home'.`

**Comment by ekohl** [comment]
In that testcase we don't include the class `pulpcore` so it can't reference the variable. That's why we explicitly set those in the tests:
https://github.com/theforeman/puppet-pulpcore/blob/9f680a141dda160d7a21f562c7392ba93afae7e8/spec/defines/admin_spec.rb#L9-L15

**Comment by ianballou** [comment]
@ekohl I think we're all good now.

File: spec/defines/admin_spec.rb [hunk 2]
```diff
@@ -11,6 +11,7 @@
           {
             pulp_settings: '/etc/pulpcore/settings.py',
             user: 'pulpcore',
+            working_dir: '/var/lib/pulp'
```
Comment (wbclark) [inline spec/defines/admin_spec.rb:14]: Hey @ianballou , could you please add to the test below, under `is_expected.to contain_exec('pulpcore-manager help')`,

`with_cwd('/var/lib/pulp')`

This way we are not only passing the param from the parent module which `pulpcore::admin` needs, we are also testing that the feature works properly at the business logic level (puppet catalogue) and continues working as expected with any future changes.

**Comment by ianballou** [comment]
@wbclark okay, I added the test condition.

**Comment by wbclark** [comment]
> @wbclark okay, I added the test condition.

Great, thank you. This looks good to me. Let's see what @ekohl says

**Review by ekohl — APPROVED** [review]
This is what I would have written as well

**Comment by goosemania** [comment]
> The change itself makes sense to me. However, please make the commit message more descriptive. Like `Always run admin cmds from Pulp's home` (not sure how much will fit). Ideally with then a body that describes that the cwd might be an unreadable directory like `/root` and dynaconf appears to unconditionally read the current directory. Pulp's user home should always be readable for the user pulp.
> 
> Also, it might be good to link to a dynaconf bugreport. IMHO it shouldn't error out on unreadable CWD.

FWIW, tried to fix, this helped https://github.com/rochacbruno/dynaconf/pull/570

**Pull request merged**